### PR TITLE
Doc update for opensuse

### PR DESCRIPTION
**Pull request by neotinker**
This is a followup to a request for CONTRIBUTING.md and REVIEW_GUIDELINES.md updates for openSUSE.

https://github.com/ros/rosdistro/pull/28385#pullrequestreview-596691495

I've also reworked the rosdep/{base,python},yaml patches to be less release specific unless necessary. I'll open separate PRs for those in smaller bunches if this one gets accepted.

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
This seems reasonable to me, but I'll wait for another review before merging.

**Review by ivanpauno — APPROVED** [review]
LGTM!

@neotinker could you resolve conflicts?

**Comment by ivanpauno** [comment]
Thanks @neotinker !

**Pull request merged**
